### ISSUE TITLE
Wrong output after creating schema with field type: dict | list

### ISSUE THREAD

**Issue by lisvv** [pydantic/pydantic#4942]
### Initial Checks

- [X] I have searched GitHub for a duplicate issue and I'm sure this is something new
- [X] I have searched Google & StackOverflow for a solution and couldn't find anything
- [X] I have read and followed [the docs](https://pydantic-docs.helpmanual.io) and still think this is a bug
- [X] I am confident that the issue is with pydantic (not my code, or another library in the ecosystem like [FastAPI](https://fastapi.tiangolo.com) or [mypy](https://mypy.readthedocs.io/en/stable))

### Description

Hello! Output for Model attribute is wrong when i use Schema with field: dict | list typing

I make simple example above in code. It's run as it is.

### Example Code

```Python
from pydantic import BaseModel


class SimpleSchema(BaseModel):
    attribute: dict | list


x = SimpleSchema(
    attribute=[{"value": 1, "foo": {'foo': 2}}, {"value": 1, "bar": {"eggs": 3}}]
)

print(x)
```
output in shell:
 attribute={'value': 'bar'}
 
 but if we make `list | dict` output will be valid

### Python, Pydantic & OS Version

```Text
pydantic version: 1.10.4
            pydantic compiled: True
                 install path: /Users/qwe/Library/Caches/pypoetry/virtualenvs/example.py-0g2MhG5m-py3.10/lib/python3.10/site-packages/pydantic
               python version: 3.10.1 (v3.10.1:2cd268a3a9, Dec  6 2021, 14:28:59) [Clang 13.0.0 (clang-1300.0.29.3)]
                     platform: macOS-13.1-arm64-arm-64bit
     optional deps. installed: ['typing-extensions']
```

### Affected Components

- [ ] [Compatibility between releases](https://pydantic-docs.helpmanual.io/changelog/)
- [ ] [Data validation/parsing](https://pydantic-docs.helpmanual.io/usage/models/#basic-model-usage)
- [ ] [Data serialization](https://pydantic-docs.helpmanual.io/usage/exporting_models/) - `.model_dump()` and `.model_dump_json()`
- [ ] [JSON Schema](https://pydantic-docs.helpmanual.io/usage/schema/)
- [ ] [Dataclasses](https://pydantic-docs.helpmanual.io/usage/dataclasses/)
- [ ] [Model Config](https://pydantic-docs.helpmanual.io/usage/model_config/)
- [X] [Field Types](https://pydantic-docs.helpmanual.io/usage/types/) - adding or changing a particular data type
- [ ] [Function validation decorator](https://pydantic-docs.helpmanual.io/usage/validation_decorator/)
- [ ] [Generic Models](https://pydantic-docs.helpmanual.io/usage/models/#generic-models)
- [ ] [Other Model behaviour](https://pydantic-docs.helpmanual.io/usage/models/) - `model_construct()`, pickling, private attributes, ORM mode
- [ ] [Plugins](https://pydantic-docs.helpmanual.io/) and integration with other tools - mypy, FastAPI, python-devtools, Hypothesis, VS Code, PyCharm, etc.

**Comment by hramezani** [comment]
Thanks @lisvv for reporting.

I can confirm the problem is reproducible on `1.10.x` but I couldn't reproduce it on the `main`.

As we are working on V2 and the plan is to just provide bugfix patches for 1.10.x, I think we can accept a fix on 1.10.x if it is an easy one.

**Comment by Kludex** [comment]
Since we can't reproduce it on V2, I'll be closing this issue.

We are still accepting PRs to fix issues on V1 (if the changes are small). 🙏
